.circleci/server-upgrade/run.sh

```diff
@@ -9,114 +14,233 @@ set -euo pipefail
 : ${HASURA_PROJECT_DIR:=$ROOT/hasura}
 : ${API_SERVER_DIR:=$ROOT/api-server}
 : ${SERVER_OUTPUT_DIR:=/build/_server_output}
 : ${SERVER_BINARY:=/build/_server_output/graphql-engine}
 : ${LATEST_SERVER_BINARY:=/bin/graphql-engine-latest}
-
+: ${HASURA_GRAPHQL_STRINGIFY_NUMERIC_TYPES:=true}
 
 LATEST_SERVER_LOG=$SERVER_OUTPUT_DIR/upgrade-test-latest-release-server.log
 CURRENT_SERVER_LOG=$SERVER_OUTPUT_DIR/upgrade-test-current-server.log
-API_SERVER_LOG=$SERVER_OUTPUT_DIR/api-server.log
 
-API_SERVER_ENDPOINT=http://localhost:$API_SERVER_PORT
 HGE_ENDPOINT=http://localhost:$HASURA_GRAPHQL_SERVER_PORT
+PYTEST_DIR="${ROOT}/../../server/tests-py"
 
-# event trigger and remote schema urls
-: ${ET_ECHO_URL:=$API_SERVER_ENDPOINT/trigger/echo}
-: ${RS_HELLO_URL:=$API_SERVER_ENDPOINT/remote-schema/hello}
+pip3 -q install -r "${PYTEST_DIR}/requirements.txt"
 
 # export them so that GraphQL Engine can use it
-export ET_ECHO_URL=$ET_ECHO_URL
-export RS_HELLO_URL=$RS_HELLO_URL
+export HASURA_GRAPHQL_STRINGIFY_NUMERIC_TYPES="$HASURA_GRAPHQL_STRINGIFY_NUMERIC_TYPES"
+# Required for testing caching
+export GHCRTS='-N1'
+# Required for event trigger tests
+export WEBHOOK_FROM_ENV="http://127.0.0.1:5592"
+export EVENT_WEBHOOK_HEADER="MyEnvValue"
+
+# graphql-engine will be run on this port
+fail_if_port_busy ${HASURA_GRAPHQL_SERVER_PORT}
+
+# Remote graphql server of pytests run on this port
+fail_if_port_busy 5000
 
 log "setting up directories"
 mkdir -p $SERVER_OUTPUT_DIR
 touch $LATEST_SERVER_LOG
 touch $CURRENT_SERVER_LOG
 
-# installing deps
-log "installing deps"
-yarn --cwd $ROOT install
-
-# updating hasura cli
-log "updating hasura cli"
-hasura update-cli
-
-# start api server for event triggers and remote schemas
-log "starting api server for triggers and remote schemas"
-yarn --cwd $API_SERVER_DIR install
-PORT=$API_SERVER_PORT yarn --cwd $API_SERVER_DIR start-prod > $API_SERVER_LOG 2>&1 &
-API_SERVER_PID=$!
-
-wait_for_port $API_SERVER_PORT
-
 # download latest graphql engine release
 log "downloading latest release of graphql engine"
-curl -Lo $LATEST_SERVER_BINARY https://graphql-engine-cdn.hasura.io/server/latest/linux-amd64
-chmod +x $LATEST_SERVER_BINARY
+download_with_etag_check 'https://graphql-engine-cdn.hasura.io/server/latest/linux-amd64' "$LATEST_SERVER_BINARY"
 
-# start graphql engine
-log "starting latest graphql engine"
-$LATEST_SERVER_BINARY serve > $LATEST_SERVER_LOG 2>&1 &
-LAST_REL_HGE_PID=$!
-
-wait_for_port $HASURA_GRAPHQL_SERVER_PORT
-log "graphql engine started"
+cur_server_version(){
+	echo "$(curl http://localhost:${HASURA_GRAPHQL_SERVER_PORT}/v1/version -q 2>/dev/null)"
+}
 
-# apply migrations
-log "applying migrations"
-hasura --project $HASURA_PROJECT_DIR migrate apply --endpoint $HGE_ENDPOINT
+log "Run pytests with server upgrade"
 
-# make a test query
-log "executing the test query"
-node $ROOT/make_test_query.js $HGE_ENDPOINT/v1alpha1/graphql
-log
+WORKTREE_DIR="$(mktemp -d)"
+rm_worktree(){
+	rm -rf $WORKTREE_DIR
```

Comment:
rm  -rf "$WORKTREE_DIR"  # with quotes

Reply:
Good catch. I have  pushed the commit with quotes.